### PR TITLE
Add list repository tags API

### PR DESCRIPTION
**Pull request by onukura**
### Before submitting a pull-request to GitBucket I have first:

- [x] read the [contribution guidelines](https://github.com/gitbucket/gitbucket/blob/master/.github/CONTRIBUTING.md)
- [x] rebased my branch over master
- [x] verified that project is compiling
- [ ] verified that tests are passing
- [x] squashed my commits as appropriate *(keep several commits if it is relevant to understand the PR)*
- [ ] [marked as closed using commit message](https://help.github.com/articles/closing-issues-via-commit-messages/) all issue ID that this PR should correct

### About this PR

Tis PR add [List repository tags API](https://docs.github.com/en/rest/reference/repos#list-repository-tags).
Maybe related to #2506 ...??

Note: [`zipball_url` and `tarball_url`](https://docs.github.com/en/rest/reference/repos#download-a-repository-archive) is not supported by GitBucket, yet I added to follow GitHub structure response.

### CODE REVIEW

File: src/main/scala/gitbucket/core/api/ApiTag.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  def apply(
+    tagName: String,
+    repositoryName: RepositoryName,
+    commitId: String
+  ): ApiTag =
+    ApiTag(
+      name = tagName,
+      commit = ApiTagCommit(sha = commitId, url = ApiPath(s"/${repositoryName.fullName}/commits/${commitId}")),
+      zipball_url = ApiPath(s"/${repositoryName.fullName}/zipball/${tagName}"),
+      tarball_url = ApiPath(s"/${repositoryName.fullName}/tarball/${tagName}")
```
Comment (takezoe) [inline src/main/scala/gitbucket/core/api/ApiTag.scala:27]: Are these URLs actually available?

Comment (onukura) [reply src/main/scala/gitbucket/core/api/ApiTag.scala:27]: @takezoe ah.. gitbucket doesn't have this urls. Ok I will make them disable!

Comment (takezoe) [reply src/main/scala/gitbucket/core/api/ApiTag.scala]: I think we can use the following URLs:
- `/${repositoryName.fullName}/archive/${tagName}.zip`
- `/${repositoryName.fullName}/archive/${tagName}.tar.gz`

Comment (onukura) [reply src/main/scala/gitbucket/core/api/ApiTag.scala]: @takezoe Thanks for the urls! I fixed PR.

**Review by takezoe — APPROVED** [review]
This looks good for now. We must not forget to update zipball_url and tarball_url when these URLs are changed.

**Comment by onukura** [comment]
wiki updated

**Pull request merged**